### DISCUSSION TITLE
Spa temperature has occasional 'spikes' and 'dips' S100103 Honolulu

### DISCUSSION THREAD

**Discussion by NZMR2** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#673] [Q&A]
Hi all,

I'll start by stating I'm pretty green when it comes to the tech details behind these units.  I guess I know enough to be dangerous!!  I've recently purchased a complete 6 wire module from Metaljay and had it delivered to me in New Zealand.  I managed to get it set up using Hardware: 6 wire 2021 (air) and DSP: 6 wire 2021 (air), PCB: V2B.  It connects and I can control via the web interface.  It's not super responsive and trying to control via the spa itself is a bit hit and miss.  The display sometimes flicks off / does a few strange behaviours.  I also can't use the timer buttons anymore. 

I managed to get it hooked up to my Home Assistant via MQTT.  I've got it showing in my main dashboard.  

For now, I'm just setting some timers to see if it's working and it seems to be.  My intent is to use some smarts to turn it on when my solar is generating excess.  However, when I reviewed the history on the 'Spa temperature', I can see that it's getting spikes (up to about 92 degrees) every few minutes.  This also shows up in the logs where it says the heater has turned off then turned on a few seconds later.  Watching the spa display it looks like this is actually happening on the spa itself too.

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/160834325/3ccb17dd-7c09-424d-8477-70f8865cb655)

Has anyone seen this behaviour before?  Any thoughts as to what to do to sort it out?  

Thanks,
Danny

**Comment by visualapproach** [comment]
The timer button may be disabled in config. The spikes is probably bit errors. Probably due to poor connection. Not necessarily bad solders but it can be too long wires or mismatched filters, faulty esp and so on.

**Reply by Rodrigocicci** [reply]
Same issue with the same pump. Tried with my module, everything works but i had this same blinking and spikes in graphic. Though it was my module and built a second one. Same issue. Got one from metaljay and same issue. Will have a week off next week and i will try re flashing the module and reconfigure HA.
I will post here if anything changes.

**Reply by visualapproach** [reply]
If using the LP filters, try without. And vice versa.

**Reply by NZMR2** [reply]
Rodrigocicci, just wondering if you've had any luck resolving this?

**Reply by Rodrigocicci** [reply]
> Rodrigocicci, just wondering if you've had any luck resolving this?

Hi NZMR2, Sorry about the delay.

Not really... I've been ignoring it until yesterday when i decided to update the firmware to the latest one. Despite the spikes in temperature and the pump switching on and imediately off once or twice a minute, I can control the spa via web ui and homeassistant. It is annoying but I survived from march till now. Not sure if i'm paying more in eletricity because of this pump flickering (probably yes).
I am in australia, pump is S100103 6wire pump and display. I tryed 2 different wemo_d1 "direct" installs and a "complete" pcb v2b from metaljay (same issue in all of them).

The problem with non responsiveness i did not fix it but got way better when I improved my network and wifi.

I would relax so much more without pump flickering and spikes on my HA graphics so i will keep trying.

Let me know if you found a solution as well.

**Comment by alexapps1** [comment]
I have exactly the same issue with S100102 pump. I tried a D1 Mini the first time and figured it was because I was out of spec, so I thought the problems were due to 3.3v vs 5v logic.

Bought the kit from metaljay, just tried it, and still get exactly the same issues.

-- Temperature up and down on control panel not working, although sounds do play.
-- When changing temperature in Web UI it seems to hunt for the temperature set.
-- Assume still getting temperature reading spikes as even the display said >90c.

I've soldered my cables on the H1B and H2B side (as per metaljay pics although he is using the old U3 rather than U2 level shifter in those pics. I assumed H1B and H2B are post resistor and capacitor which I figured are there to help stabilize things.

There is someone selling one on eBay with cables soldered on the H1A and H2A side. I don't know if this is the issue, maybe I should have done this?

**Reply by alexapps1** [reply]
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/26008808/f6b55112-5ca6-4553-b7fa-f2df120ea0e7

Change temp to 29c at 00:03
Target changes to 31c at 00:11
Target changes to 21c at 00:22
Target changes to 26c at 00:30
Target changes to 28c at 00:33
Target changes to 36c at 00:41
Target changes to 34c at 00:43
Target changes to 32c at 00:46
Target changes to 30c at 00:50
Target changes to 29c at 00:50

Finally sticks at 29c.

**Reply by visualapproach** [reply]
The signals from cio seems to be bad. It occasionally gets proper messages (incl temp) but then it has already passed the target so it sends the opposite button (up or down) and only settles when it by chance gets the right value.
I reckon you don't want to do soldering since you bought it pre made. Otherwise I would recommend trying as short cables as possible and try with the A ports.

**Reply by alexapps1** [reply]
Thanks @visualapproach. 

Sorry, I should have been clear, I bought the parts but soldered myself. I will try short cables on the A ports now, and shorten the cables on the B ports if that still doesn't work.

**Reply by alexapps1** [reply]
Shortened cables to 1 inch, using A ports and no change.

Very strange that whatever I change I seem to consistently get the same behaviour. 

Being in the UK with 50hz electricity wouldn't change anything clock related would it? 

I can try shorter cables on B port unless you can think of anything else to try first to save me unsoldering those!

**Reply by metaljay** [reply]
Sorry I missed this, have you configured your pump and board revision correctly? Have you tried experimenting with other options? I ask as the fact the display isn’t working is either the red LLC or pump config

**Reply by alexapps1** [reply]
Yes I think so ![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/26008808/42809384-11e2-4bf0-9d0e-b2045161d21a)

It's all working, it's just like there is noise on the data line to the pump, just misses messages or sends false messages back. 

Only thing I haven't tried is shortening the cables on the 'B' side as I needed to put it back together to use it. 

Shortening cables on A side made no difference, moving from D1_Mini to full recommended setup made no difference. 

Cables being nowhere near anything else, and pump/heater being off made no difference, which is why I'll be surprised if it is anything to do with cable length but I've been wrong before...!

**Reply by alexapps1** [reply]
@visualapproach @metaljay I tried resoldering all connections and updating to latest development version in case it was linked to the other thread where someone was having communication issues. I get the following in debug, all still the same symptoms. 

2024-07-18 14:00:43UTC.  SW:2024-07-18-0900 
CIO-ESP:C0,1,FE,B7,FE,F7,FE,1,FF,F5,FF
DSP-ESP:1B,1B,0,0,0,0,0,0,0,0,0
ESP-CIO:1B,1B,0,0,0,0,0,0,0,0,0
ESP-DSP:C0,1,FE,B7,FE,F7,FE,1,FF,F5,FF
CIO msg count: 43 DSP msg count: 1299
CIO msg quality: 22.000000% DSP msg quality: 100.000000%

Packet errors:1
bit0-bit error, bit1-too many bytes, bit2: too few bytes.

When I looked at /inputs/ I noticed there is about 8us between the entries, which seems to be quite different to the 20 - 25 that other people had?

**Reply by visualapproach** [reply]
See if this makes any difference. In main.cpp line 330 put this code:
`    WiFi.mode(WIFI_STA); WiFi.setOutputPower(15.0);`
You can set it in 0.25 steps between 1 and 20.5. I was thinking maybe the wifi is making the noise.

**Reply by alexapps1** [reply]
Even with: `WiFi.setOutputPower(1.0);`
Still no change.

**Reply by alexapps1** [reply]
With [this set up](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/615#discussioncomment-10035586), can I connect the board via USB and run the hardware test? The manual says not to unless the 5v is disconnected, but I don't know if it is or not.

These pictures were prior to resoldering. I'd be surprised if my soldering is consistently bad enough that I cause the same issues with this board (soldered twice), and the d1_mini I had before - I guess I can't rule it out though.

**Reply by visualapproach** [reply]
It's okay to use only USB. Not both USB and extra 5V supply

**Reply by visualapproach** [reply]
> Even with: `WiFi.setOutputPower(1.0);` Still no change.

good to know at least

**Reply by alexapps1** [reply]
The tests passed. 
![image](https://github.com/user-attachments/assets/a685a88b-1fb2-4157-869b-3ad6d4a51eaf)

**Reply by visualapproach** [reply]
Good. The test is at a lower frequency than the real signals but it seems it's good. See if it works with only resistors. I.e. remove the caps.

**Reply by alexapps1** [reply]
Removal of caps seemingly has made no difference. The few messages I got logged when using debug-on and pressing a button, now say ~57% message quality which is higher before. But I assume that needs to be 100% for it to work.

**Reply by visualapproach** [reply]
Ideally it would be 100% but it doesn't have to be. It means that the ESP just ignores some messages, and the data is probably the same in the next transmission. But it could give those spikes if looking at the graphs

**Reply by alexapps1** [reply]
One of the main issues is the button presses on the display don't really work either. The chirp works 100%, but the message don't seem to get to the CIO side reliably. If you hold temp down, it goes chirp chirp chirp chirp and eventually lowers the temperature. Same as using the web interface, it hunts for the target temperature for some time.

**Reply by visualapproach** [reply]
Yes that's because too many messages are lost/ignored between esp and cio.

**Reply by visualapproach** [reply]
Can you drag the file log.txt here? Preferably as a file, not just text, to make it easier to scroll the thread on the phone

**Reply by alexapps1** [reply]
[log-nocaps.txt](https://github.com/user-attachments/files/16286733/log-nocaps.txt)

**Reply by visualapproach** [reply]
Don't know if the poor signal is due to speed or electronics. Guess the latter. Maybe the circuit requires another configuration. Maybe you or someone knows what chips are used in the dsp/cio?

You could also try other pins. Like swapping cio side with dsp side and reconfigure a custom pcb in hwconfig.

**Reply by alexapps1** [reply]
![image](https://github.com/user-attachments/assets/b1601ec7-19bd-4305-85b7-0e25c9614f30)

Which chip do we need to identify?

**Reply by visualapproach** [reply]
I guess this one. 
![image](https://github.com/user-attachments/assets/6f66f99d-3cab-4076-9a4f-65b7c98ca236)

But it would be better to look at the display PCB. Since we want to mimic the display end of the CIO cable. Not sure it will help, but cannot hurt.

**Reply by alexapps1** [reply]
![image](https://github.com/user-attachments/assets/c4ee24ec-8304-4783-9aed-cbc502417cf2)

**Reply by visualapproach** [reply]
Nice picture. I didn't get any clues from the datasheet. Maybe it would make a difference with another LLC. Blue vs red. But I don't know.

**Reply by alexapps1** [reply]
@visualapproach 
These are the two chips in the CIO

![image](https://github.com/user-attachments/assets/8ff8ece8-6d1f-40c1-8810-d993faa328d5)

![image](https://github.com/user-attachments/assets/52866e41-6a0d-4ca5-af2b-8c50a6a72925)

**Reply by visualapproach** [reply]
The upper picture is the communication chip (1628) the lower is a led driver chip. You wrote that they are in the CIO but you meant display? It looks like they have 100 ohms resistors (101=one, zero and one zero) and unknown capacitors. Maybe better signal if using 100ohms resistors?

**Reply by alexapps1** [reply]
Forgot to update, tried with the 100 ohm resistors on the display side - no change.

**Comment by PokoLoko2014** [comment]
Have you made any progress with the temperature problem? I have exactly the same problem with the S100101 pump, V2B board.

**Reply by PokoLoko2014** [reply]
![Screenshot_20240825_120356_Home Assistant](https://github.com/user-attachments/assets/5b91cf8e-420f-4f40-a990-3201e8269abb)

**Reply by PokoLoko2014** [reply]
![20240823_191232](https://github.com/user-attachments/assets/f9814123-a0f8-4eca-8e28-b23e73e4f257)

**Comment by visualapproach** [comment]
I'd try with 100 ohms resistors and no caps.

**Reply by PokoLoko2014** [reply]
Like this?

![20240825_131012](https://github.com/user-attachments/assets/509d513b-7e45-4405-8acc-7c833a01ab55)

**Reply by PokoLoko2014** [reply]
With the resistors the display doesn't work at all and it doesn't respond via the web UI either.
I then thought that instead of the caps, a bridge would have to be put in to provide a connection to the resistors. But this version didn't work in the end either, I couldn't even access the web UI.
![20240825_164945](https://github.com/user-attachments/assets/b764abe6-8e70-4750-ba52-fdd9a1633911)

**Reply by visualapproach** [reply]
Well then you shorted all lines to GND. Hope you didn't damage the pump.
The display could be left as is on A side if it worked before.
If that doesn't work I dunno.

**Reply by PokoLoko2014** [reply]
Yes, I connected it afterwards but unfortunately without any improvement

![temp](https://github.com/user-attachments/assets/766e5760-9662-4039-9e27-4b9fe78472f4)

**Comment by jukkapirinen** [comment]
I have used d1 mini and I have noticed that at the same time when there is temp spikes down or up, the F/C unit gets briefly toggled C - F - C. Even though no one toggled that, so maybe incorrect readings from the pump. I have the button disabled in the config. I have had the spikes only couple times a week. Also have had incorrect readings for bubbles too, maybe once a week. Often same time when also have temp spikes.

Today I have installed V2B with red LLC and caps, without resistors. Right after installed the new device, I toggled manually the C - F - C and got the temp spike once at that time. So maybe somehow related?

Using 6 wire 2021 config.
Pump is S100101 / P05326

**Comment by PokoLoko2014** [comment]
It can't be that other people with the same components and the same pump have no problems at all. Can someone with the same pump please contact us and tell us what hardware they are using.
Pump: S100101

**Comment by jukkapirinen** [comment]
I would check that the soldering is good and that there are no loose strands in the wire that might short with nearby wires.

I haven't had single spike in temp since installed the new board, but that is only 3 days ago. Except the one I got after toggled c-f-c right after install. 

![image](https://github.com/user-attachments/assets/e8a47f64-cf47-4e20-9882-9a5dc1d362d2)

Here is photo of my board.
![20240829_211451](https://github.com/user-attachments/assets/f4274579-a835-4c85-859d-e2f01fa5c7ce)
Caps are 220pF MLCC Monolithic Ceramic Capacitors.
I used female headers for the ESP and LLC, but those were shitty to solder, even I used flux and good solder.


Edit: do you have  this enabled in platformio.ini?
    -DFILTER_6W_SPIKES=1

**Reply by PokoLoko2014** [reply]
Thanks, I don't have the capacitors in there, but I'll try with them 👍

**Reply by PokoLoko2014** [reply]
"-DFILTER_6W_SPIKES=1" was enabled had not changed anything in the configuration.
The caps are now in, the spikes look better but still there.

![image](https://github.com/user-attachments/assets/ffc46cee-8630-43eb-8f88-ae3e5859b538)

**Reply by PokoLoko2014** [reply]
@jukkapirinen
edit: the caps and the new flash have did not help. I give up I feel like I've been through everything now

I'll flash the esp again to make sure something didn't go wrong.
Does your configuration look the same?
![image](https://github.com/user-attachments/assets/7f021b0e-2af7-4c76-8bb9-e1f6d44e0c89)

**Reply by jukkapirinen** [reply]
Yes the settings are same for me.

**Comment by diegoschlauri** [comment]
Hi
I've got absolutely the same problem with the spikes.
![image](https://github.com/user-attachments/assets/6f421611-2480-448a-8611-49052f2ed3d5)
I've installed the board this week with the newest version and a PCB_V2B and the red LLC.
I think these are not random spikes due connection issues. The spikes are the exact fahrenheit from the configurated celsius.
Another clue: the temps in fahrenheit and celsius are the same...
![image](https://github.com/user-attachments/assets/42dcb40f-38c1-49e3-b1eb-cdd1d60b50e8)

Best regards Diego

**Reply by PokoLoko2014** [reply]
Yes, you're right, Celsius and Fahrenheit are exactly the same values.

Are you also unable to adjust the temperature properly via the display?

![Screenshot_20240907_073612_Home Assistant](https://github.com/user-attachments/assets/a455dab6-f441-4c3a-8c1f-8118ba5996a0)

**Reply by PokoLoko2014** [reply]
In the web UI it is displayed correctly but nothing changes in Home Assistant. What is also noticeable is that the spikes are the values ​​in Fahrenheit.

![Screenshot_20240907_074723_Chrome](https://github.com/user-attachments/assets/6fdd41e5-b512-4603-a7c4-69e2d8c6fc16)
![Screenshot_20240907_074903_Home Assistant](https://github.com/user-attachments/assets/85713f85-d4bd-4ee8-8be1-d1f29fe64889)

**Reply by jukkapirinen** [reply]
Home assistant converts the F to C so that's why it displays the same. However in the esp software the conversion seems to be briefly switched way around or wrong. Maybe because of a bug or some interference with the F/C status.  Eg pump is set to F but it still displays in C or another way around. 

37'F = 2.7'C
37'C = 98.6 'C

So this seems to reflect the spikes.

**Reply by visualapproach** [reply]
It could be that the unit bit flips occasionally due to noisy signal from the CIO. The ESP will then think the shown temperature is in the other format.

**Reply by diegoschlauri** [reply]
Yes, the temperatur ajustment is also a little flimsy. When I change the target temperature(on homeasstiant aswell), the target temp moves around and settles in after some seconds. Easy fix could be to ignore the unit, and set it hardcoded via the settings?

**Reply by alexapps1** [reply]
> Home assistant converts the F to C so that's why it displays the same. However in the esp software the conversion seems to be briefly switched way around or wrong. Maybe because of a bug or some interference with the F/C status.  Eg pump is set to F but it still displays in C or another way around. 
> 
> 
> 
> 37'F = 2.7'C
> 
> 37'C = 98.6 'C
> 
> 
> 
> So this seems to reflect the spikes.

I think is because it is randomly seeing phantom button presses. One of those being the c/f switch button.

**Reply by jukkapirinen** [reply]
Yes it's some signal noise, at the same time it might show that bubbles were on and then right away off. 
Disabling C/F button from the app have no effect.

When changed the Wemos D1 mini to NodeMCU + board + LLC  + capacitors, almost all the spikes went away, getting spike maybe once per week anymore.

**Reply by alexapps1** [reply]
What value capacitors and resistors did you use on which model pump and display?

My Home Assistant logs are crazy, showing on/off hundreds of times a day. Tried different wires, different resistors, with and without resistors and capacitors. No real change. Slightly worried it's going to knacker the pump, but it hasn't yet.

**Comment by alexapps1** [comment]
Has anyone tried with the blue LLC instead?

**Comment by visualapproach** [comment]
There are 3 temperature sensors. TMP, TMPC, TMPF. If the unit bit is flipping, then TMPC and TMPF will give spikey graphs. The TMP (raw value from the pump) would be different. Possibly good. Can you check?

**Reply by onemorename** [reply]
Any update on this? What's the latest status on this issue?

**Reply by PokoLoko2014** [reply]
> Any update on this? What's the latest status on this issue?

No, still the same problem.

**Reply by beegmouse** [reply]
The unit I bought was pre-assembled off ebay.

![image](https://github.com/user-attachments/assets/3965d2f9-2e65-457f-9125-b7553479d59e)

No filters caps or resistors.

Am I right in thinking a suggested hardware fix is moving the connectors.
![image](https://github.com/user-attachments/assets/c537fbc6-75d4-4e7d-bb2a-acb90c089180)
Adding 100ohm resistors and 220pF

I do wonder if the ESP could run a rolling 3 second average and crop values which are 10 degrees greater/lower than the current avereage.

This wouldn't fix what is displayed, but would stop erroneous automations and smooth graphs.
